merkle/kvpairs.go

```diff
@@ -0,0 +1,49 @@
+package merkle
+
+import (
+	"sort"
+
+	wire "github.com/tendermint/go-wire"
+	"golang.org/x/crypto/ripemd160"
+)
+
+// KVPair is a list of KVPairs that is hashed via `SimpleHashFromHashables`.
+// NOTE: Behavior is undefined with dup keys.
+type KVPair struct {
+	Key   string
+	Value interface{}
+}
+
+func (kv KVPair) Hash() []byte {
+	hasher, n, err := ripemd160.New(), new(int), new(error)
+	wire.WriteString(kv.Key, hasher, n, err)
+	if kvH, ok := kv.Value.(Hashable); ok {
+		wire.WriteByteSlice(kvH.Hash(), hasher, n, err)
+	} else {
+		wire.WriteBinary(kv.Value, hasher, n, err)
+	}
+	if *err != nil {
+		panic(*err)
+	}
+	return hasher.Sum(nil)
+}
+
+type KVPairs []KVPair
+
+func (kvps KVPairs) Len() int           { return len(kvps) }
+func (kvps KVPairs) Less(i, j int) bool { return kvps[i].Key < kvps[j].Key }
+func (kvps KVPairs) Swap(i, j int)      { kvps[i], kvps[j] = kvps[j], kvps[i] }
+func (kvps KVPairs) Sort()              { sort.Sort(kvps) }
```

Comment:
I just tagged you on the internal thread about the support for Go1.8+ where we'll later on just use `sort.Slice` that is
```go
sort.Slice(kvPairs, func(i, j int) bool {
   kvi, kvj := kvPairs[i], kvPairs[j]
   return kvi.Key < kvj.Key
})
```

Reply:
iavl only compiles with go 1.8+
I questioned this, and bucky, alexis, and adrian all spoke in favor of making this a requirement. It is commented in cosmos-sdk (due to iavl dependency), let's just update everything to 1.8